File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    long long lnum = num;
    int cnt = 0;
    while(cnt != 500)
    {
        if(lnum == 1) break;
        
        if(lnum%2 == 0) lnum/=2;
        else lnum = lnum*3+1;
        
        cnt++;
    }
    if(cnt < 500) return cnt;
    else return -1;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cs
public class Solution {
    public int solution(int num) {
        int answer = 0;
        long max = (long)num;
        while(max != 1)
        {
            if(answer ==500)
            {
                answer = -1;
                break;
            }
            if(max %2 == 0) max/=2;
            else if(max %2 != 0) max = max*3+1;
            answer++;
        }
        
        
        return answer;
    }
}